swarm/network/simulations/discovery/dynamic_test.go

```diff
@@ -0,0 +1,359 @@
+package discovery
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/simulations"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/network"
+)
+
+const (
+	bootNodeCount             = 3
+	quarantineNodeCount       = 5
+	defaultHealthCheckRetires = 10
+	defaultHealthCheckDelay   = time.Millisecond * 250
+)
+
+var (
+	upNodes     []*discover.NodeID
+	upNodesLast int
+	mu          sync.Mutex
+	bootNodes   []*discover.NodeID
+	rpcs        map[discover.NodeID]*rpc.Client
+	subs        map[discover.NodeID]*rpc.ClientSubscription
+	events      chan *p2p.PeerEvent
+	ids         []discover.NodeID
+)
+
+func TestDynamicDiscovery(t *testing.T) {
+	t.Run("10/10/sim", dynamicDiscoverySimulation)
+}
+
+func dynamicDiscoverySimulation(t *testing.T) {
```

Comment:
I like short descriptions of what the test/simulation does. 
E.g. "First set up network, start fixed number of nodes, start/stop dynamically nodes" (whatever it is really doing)